lib/syndicates.js

```diff
@@ -0,0 +1,257 @@
+'use strict';
+
+const fetch = require('node-fetch');
+const cheerio = require('cheerio');
+const { hash } = require('./utils');
+
+// Define mappers for mapping html elements to drop data
+// Default mapper, maps the most syndicate ware lists.
+function defaultMapper($, element, section) {
+  // Name is mostly located within an "a" tag, sometimes within a "strong" instead.
+  const name = $(element).find('div:last-child a span').text() || $(element).find('div:last-child strong span').text();
+  const value = $(element).find('p span').text();
+
+  // Sometimes the rank information is in the fourth div, so fallback to that.
+  // In that case the third div seems to be empty.
+  const rank = $(element).find('div:nth-of-type(3)').text() || $(element).find('div:nth-of-type(4)').text();
+  return {
+    _id: hash(name),
+    item: name,
+    chance: 100,
+    rarity: `${rank} (${value} standing)`,
+    place: section,
+  };
+}
+
+// Some syndicate pages display ware lists as a gallery, for example "Grandmother" for Entrati.
+function galleryMapper($, element, section) {
+  // Text contains something like "Deimos Vault Scene50  Grandmother Token"
+  const text = $(element).find('.lightbox-caption').text();
+
+  // Search for the first index of a numeric value.
+  const numberIndex = text.search(/(?=\d)/g);
+
+  // Split the text and extract the name from it.
+  const name = text.slice(0, numberIndex);
+
+  // The remainder contains the cost information.
+  // Remove any unicode and make sure there are no duplicate spaces.
+  const value = text.slice(numberIndex).replace(/[^\\x00-\x7F]/g, ' ').replace('  ', ' ');
+  return {
+    _id: hash(name),
+    item: name,
+    chance: 100,
+    rarity: value,
+    place: section,
+  };
+}
+
+// Custom mapper for the Archimedean vendor for The Holdfasts.
+function archimedeanMapper($, element, section) {
+  const name = $(element).find('div:nth-of-type(3) > a > span').text();
+  const value = $(element).find('div:nth-of-type(1) > div > a > span').text();
+  return {
+    _id: hash(name),
+    item: name,
+    chance: 100,
+    rarity: `${value} Thrax Plasm`,
+    place: section,
+  };
+}
+
+// Custom mapper for the Operational Supply syndicate.
+function operationalSupplyMapper($, element, section) {
+  const name = $(element).find('div:last-child a span').text();
+  // We remove any unicode from the text.
+  const standing = $(element).find('> div:nth-of-type(1) > div:nth-of-type(1) .sortkey').text().replace(/[^\\x00-\x7F]/g, ' ');
+  const credits = $(element).find('> div:nth-of-type(1) > div:nth-of-type(2) .sortkey').text().replace(/[^\\x00-\x7F]/g, ' ');
+  const rank = $(element).find('> div:nth-of-type(2)').text();
+  return {
+    _id: hash(name),
+    item: name,
+    chance: 100,
+    rarity: `${rank} (${standing} / ${credits})`,
+    place: section,
```

Comment:
```suggestion
    rarity: 'Common',
    place: `${syndicate} (${standing})`,
    standing,
    credits,
```

the last 2 will be filtered out by the slim wrapper, but anyone needing them for their purposes will have them readily available

a similar mapping should be used for the others.

Reply:
Done!